sklearn/cluster/tests/test_k_means.py

```diff
@@ -44,15 +44,6 @@
 X_csr = sp.csr_matrix(X)
 
 
-def test_kmeans_dtype():
-    rnd = np.random.RandomState(0)
-    X = rnd.normal(size=(40, 2))
-    X = (X * 10).astype(np.uint8)
```

Comment:
Do we have other tests that ensure it can accept unusual dtype and produce meaningful output? Maybe this test needs to be retained, but without the warning check.